### PR TITLE
[Tests-Only] Convert drone.yml to drone.star

### PR DESCRIPTION
**Pull request by HariBhandari07**
### Description
Convert `.drone.yml` to `.drone.star`

### Related Issues
On top of https://github.com/owncloud/owncloud-sdk/pull/543
- Fixes https://github.com/owncloud/owncloud-sdk/issues/544

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		'master'
+	],
+
+	'yarnlint': True,
+
+	'build': True
+}
+
+def main(ctx):
+    return [ buildDocs() + buildSystem() + installServer() + configureServer() + owncloudLog() + waitForOwncloud() + prepareTestConfig() + test() + codecov() ]
```
Comment (dpakach) [inline .drone.star:14]: This should return the array of pipelines, not the steps

File: .drone.star [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+			'/usr/local/bin/apachectl',
+			'-e',
+			'debug',
+			'-D',
+			'FOREGROUND'
+		]
+	}]
+
+
+def fullBuildAndPublish():
```
Comment (dpakach) [inline .drone.star:220]: This is not used anywhere. use this in main function.

File: .drone.star [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+		'master'
+	],
+
+	'yarnlint': True,
+
+	'build': True
+}
+
+def main(ctx):
+    return [ fullBuildAndPublish() + testWithinSubFolder() ]
```
Comment (phil-davis) [inline .drone.star:14]: Might need to be:
```suggestion
    return [ fullBuildAndPublish(), testWithinSubFolder() ]
```

File: .drone.star [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            'branch': 'master'
+        },
+        'steps':
+            incrementVersion() +
+            buildDocs() +
+            buildSystem() +
+            prepareTestConfig() +
+            test() +
+            publishDocs() +
+            publishSystem()
```
Comment (individual-it) [inline .drone.star:147]: this

File: .drone.star [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    return [{
+        'name': 'test',
+        'image': 'owncloudci/chromium',
+        'pull': 'always',
+        'commands': [
+            'yarn test-drone'
+        ]
+    }]
+
+def publishDocs():
```
Comment (individual-it) [inline .drone.star:79]: this step will run twice, in fullBuildAndPublish and testWithinSubFolder, it should only run only once after both types of tests pass

File: .drone.star [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+            }
+        },
+        'when': {
+            'event': [
+                'push'
+            ]
+        }
+    }]
+
+def publishSystem():
```
Comment (individual-it) [inline .drone.star:102]: same as publishDocs

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by LukasHirt** [comment]
As we're switching to test with pact.io does it still make sense to run the tests in subfolder? Since we mock the responses there won't be much difference between those two anymore, will be?

**Comment by individual-it** [comment]
@LukasHirt I think it makes sense as we are not testing the server but the consumer here. So if there is a chance that the SDK has a bug and reacting differently when sending a requests to `/sub/ocs/v1.php...` and not to `ocs/v1.php....` then it does make sense.
I believe we had such bugs in oc10. The backend would deliver correctly but the UI would not display the result correctly e.g. because its parsed the path wrong.
CC @PVince81 we talked about that in the chat recently

**Comment by LukasHirt** [comment]
> I think it makes sense as we are not testing the server but the consumer here. So if there is a chance that the SDK has a bug and reacting differently when sending a requests to /sub/ocs/v1.php... and not to ocs/v1.php.... then it does make sense.

I see. Thanks for clearing out!

File: .drone.star [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+            'event': [
+                'push'
+            ]
+        }
+    }]
+
+def fullBuildAndPublish():
+    return {
+        'kind': 'pipeline',
+        'name': 'Full build and publish',
```
Comment (individual-it) [inline .drone.star:128]: this pipeline does not publish anymore

File: .drone.star [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+            },
+        },
+        'when': {
+            'event': [
+                'push'
+            ]
+        }
+    }]
+
+def fullBuildAndPublish():
```
Comment (individual-it) [inline .drone.star:125]: this pipeline does not publish anymore

**Review by individual-it — CHANGES_REQUESTED** [review]

File: .drone.star (hunk 7 above)
Comment (dpakach) [reply .drone.star:128]: That will only run after merge so we need to merge this Pr to see if that works.

**Review by individual-it — APPROVED** [review]

**Comment by individual-it** [comment]
CC @LukasHirt I hope the publishing works

**Comment by dpakach** [comment]
Publishing doesnt works currently.
Fixed in https://github.com/owncloud/owncloud-sdk/pull/561

**Pull request merged**